lib/elastic_apm/spies/shoryuken.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module ElasticAPM
+  # @api private
+  module Spies
+    # @api private
+    class ShoryukenSpy
+      # @api private
+      class Middleware
+        def call(worker_instance, queue, sqs_msg, body)
+          transaction = ElasticAPM.start_transaction(
+            job_class(worker_instance, body),
+            'shoryuken.job'
+          )
+
+          ElasticAPM.set_label('shoryuken.id', sqs_msg.message_id)
+          ElasticAPM.set_label('shoryuken.queue', queue)
+          if sqs_msg.respond_to?(:attributes)
+            ElasticAPM.set_label('shoryuken.attributes', sqs_msg.attributes)
+          end
+          ElasticAPM.set_label('shoryuken.body', body)
```

Comment:
What is the nature of `.attributes` and `.body`? Do they fit labels (indexed, max 1024 chars)?

Perhaps `set_context` is better for attributes? Perhaps body is better omitted?

Reply:
They both contain a dictionary of metadata. Some information seemed useful, but they can be omitted in this contribution. For some reason, there are not serialized as JSON but rather as string representation of a ruby dict.

`body`
```ruby
{"job_class"=>"SomeJobClass", "job_id"=>"6a9aefe8-fcf1-4878-b841-b5221bbc7546", "provider_job_id"=>nil, "queue_name"=>"some_queue_name", "priority"=>nil, "arguments"=>["1234"], "executions"=>0, "exception_executions"=>{}, "locale"=>"en", "timezone"=>"UTC", "enqueued_at"=>"2019-10-12T10:07:41Z"}
```

`attributes`:
```ruby
{"ApproximateFirstReceiveTimestamp"=>"1576145416", "SenderId"=>"queue", "ApproximateReceiveCount"=>"1", "SentTimestamp"=>"1576145416"}
```
 

Reply:
Are they strings before setting as label too? Labels will `to_s` on anything but Numbers and Booleans.

I think we should omit them by default. Then we won't have to worry about stripping sensitive information and such neither. Perhaps mention somewhere in the docs how to include them manually? _wdyt?_

Reply:
ok, omitting them seems to be the best option for now